client/sites/tools/logs/index.tsx

```diff
@@ -251,3 +263,257 @@ export const SiteLogs = ( {
+	const [ severity ] = useState( () => {
+		return getFilterQueryParam( 'severity' ) || '';
+	} );
+	const [ requestType ] = useState( () => {
+		return getFilterQueryParam( 'request_type' ) || '';
+	} );
+
+	const [ requestStatus ] = useState( () => {
+		return getFilterQueryParam( 'request_status' ) || '';
+	} );
+
+	const { data, isFetching } = useSiteLogsQuery( siteId, {
+		logType,
+		start: dateRange.startTime.unix(),
+		end: dateRange.endTime.unix(),
+		filter: buildFilterParam( logType, severity, requestType, requestStatus ),
+		sortOrder: view.sort?.direction,
+		pageSize: view.perPage,
+		pageIndex: view.page,
+	} );
+
+	return {
+		data: data?.logs ? data.logs : EMPTY_ARRAY,
+		paginationInfo: {
+			totalItems: data?.total_results || 0,
+			totalPages:
+				!! data?.total_results && !! view.perPage
+					? Math.ceil( data.total_results / view.perPage )
+					: 0,
+		},
+		isLoading: isFetching,
+	};
+};
+
+const useFields = ( { logType }: { logType: LogType } ) => {
+	if ( logType === 'php' ) {
+		return [
+			{ id: 'severity', label: 'Severity', enableSorting: false },
+			{ id: 'timestamp', label: 'Timestamp' },
+			{ id: 'message', label: 'Message', enableSorting: false },
+			{ id: 'kind', label: 'Kind', enableSorting: false },
+			{ id: 'name', label: 'Name', enableSorting: false },
+			{ id: 'file', label: 'File', enableSorting: false },
+			{ id: 'line', label: 'Line', enableSorting: false },
+		];
+	}
+
+	// TODO: translate
+	return [
+		{ id: 'request_type', label: 'Request type', enableSorting: false },
+		{ id: 'date', label: 'Date' },
+		{ id: 'status', label: 'Status', enableSorting: false },
+		{ id: 'request_url', label: 'Request URL', enableSorting: false },
+		{ id: 'timestamp', label: 'Timestamp', enableSorting: false },
+		{ id: 'body_bytes_sent', label: 'Body bytes sent', enableSorting: false },
+		{ id: 'cached', label: 'Cached', enableSorting: false },
+		{ id: 'http_host', label: 'HTTP Host', enableSorting: false },
+		{ id: 'http_referer', label: 'Referrer', enableSorting: false },
+	];
+};
+
+const getVisibleFieldsForLogType = ( logType: LogType ) => {
+	if ( logType === 'php' ) {
+		return [ 'severity', 'timestamp', 'message' ];
+	}
+	return [ 'request_type', 'date', 'status', 'request_url' ];
+};
+
+const getSortFieldForLogType = ( logType: LogType ) => ( logType === 'php' ? 'timestamp' : 'date' );
+
+export const SiteLogsDataViews = ( { logType }: { logType: LogType } ) => {
+	// TODO:
+	// - DataViews:
+	//   - filters
+	//   - search
+	//   - max width for long cells
+	//   - spacing left/right
+	// - Fields:
+	//   - timestamp for web logs
+	// - Address the "show more" interaction.
+
+	const { __ } = useI18n(); // Can we use translate instead?
+
+	const moment = useLocalizedMoment();
+	const getLatestDateRange = useCallback( () => {
+		const startTime = moment().subtract( 7, 'd' );
+		const endTime = moment();
+		return { startTime, endTime };
+	}, [ moment ] );
```

Comment:
I don't want to make your life harder. But@wordpress/dataviews is 340KB gzipped and moment is 73KB gzipped. It would be amazing if we follow up with some performance considerations (eg async loading, not using moment, etc..).